multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
+
+                    if os.path.exists(safe_save_path) == True:
+                        print("20x already moved")
+
+                    elif os.path.exists(safe_save_path) == False:
+                        image = urllib.URLopener()
+                        image.retrieve(full_url, safe_save_path)
+                        print("20x image moved")
+                    self.image_20 = safe_save_path
+                        
+                else:
+                    raise Exception("Couldn't find image in LIMS. Check you have the selected the correct slice.")
+                
+            except KeyError:
+                raise Exception('No Slice Selected')
+        else:
+            raise Exception('No Slice Selected')
+       
+
+
+    def create_lims_json(self):
+        """
+        Checks that markers are loaded into the canvas
+        Gets timestamp
+        Creates json for each cell clusters and saves locally
+        copies json to correct incoming folder
+        creates trigger file and drops in correct incoming folder
+        """
+        
+        items = self.mosaic_editor.canvas.items 
+        markers = [i for i in items if isinstance(i, MarkersCanvasItem)]
+        if len(markers) != 1:
+            raise Exception("Must have exactly 1 Markers item in the canvas.")
+        
+        for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
+                        image = i
+                except AttributeError:
+                    pass
+
+        sites = set()
+        for item in items:
+            handle = item.opts.get('handle')
+            if handle is None:
+                continue
+            parent = handle.parent()
+            if parent.shortName().startswith('site_'):
+                sites.add(parent)
+
+        if len(sites) == 0:
+            raise Exception("No files loaded from site; can't determine which site to submit.")
+        if len(sites) > 1:
+            raise Exception("Files loaded from multiple sites; can't determine which site to submit.")
+        self.site_dh = list(sites)[0]
+        ts = self.site_dh.info()['__timestamp__']
+
+        
+        cluster_id = []
+        clusters = lims.cell_cluster_ids(self.slice_id)
+        for cid in clusters:
+            cluster_tmsp = lims.specimen_metadata(cid)['acq_timestamp']
+            if cluster_tmsp == ts:
+                cluster_id = cid
+                cluster_name = lims.specimen_name(cid)
+                print(cluster_name)
+            
+
+        if cluster_id == []:
+            raise Exception("Couldn't match Multipatch Log to data in LIMS")
+          
+        #convert the pipette.yml file into a usable dictionary
+        pipette_path = os.path.join(self.site_dh.name(), 'pipettes.yml')
+
+        if os.path.exists(pipette_path) == False:
+            raise Exception("Could not find pipette log")
+
+        with open(pipette_path, 'r') as the_file:
```

Comment:
Don't parse this yourself, use `yaml.load`

Reply:
that is very useful and so much better than what I was doing